Microsoft.HealthVault.Fhir/Transformers/AllergyToFhir.cs

```diff
@@ -0,0 +1,134 @@
+
+            if (allergy.AllergenType != null)
+            {
+                allergyInTolerance.SetAllergyIntoleranceCategory(allergy.AllergenType, allergyExtension);
+            }
+
+            if (allergy.Reaction != null)
+            {
+                allergyInTolerance.Reaction = new List<AllergyIntolerance.ReactionComponent>
+                {
+                     new AllergyIntolerance.ReactionComponent {
+                         Manifestation = new List<CodeableConcept>{ allergy.Reaction.ToFhir() },
+                         Description = allergy.Reaction.Text
+                     }
+                };
+            }
+
+            if (allergy.TreatmentProvider != null)
+            {
+                allergyInTolerance.AddAsserter(allergy.TreatmentProvider.ToFhir());
+            }
+
+            if (allergy.Treatment != null)
+            {
+                allergyExtension.AddExtension(HealthVaultExtensions.AllergyTreatement, allergy.Treatment.ToFhir());
+            }
+
+            if (allergy.AllergenCode != null)
+            {
+                allergyExtension.AddExtension(HealthVaultExtensions.AllergenCode, allergy.AllergenCode.ToFhir());
+            }
+
+            if (allergy.IsNegated != null)
+            {
+                if (allergy.IsNegated.Value)
+                {
+                    allergyInTolerance.ClinicalStatus = AllergyIntoleranceClinicalStatus.Resolved;
+                }
+                else
+                {
+                    allergyInTolerance.ClinicalStatus = AllergyIntoleranceClinicalStatus.Active;
+                }
+            }
+
+            if (allergy.CommonData != null)
+            {
+                var note = new Hl7.Fhir.Model.Annotation();
+                note.Text = allergy.CommonData.Note;
+                allergyInTolerance.Note = new List<Hl7.Fhir.Model.Annotation> { note };
+            }
+
+            allergyInTolerance.Type = AllergyIntoleranceType.Allergy;
+            allergyInTolerance.Extension.Add(allergyExtension);
+
+            return allergyInTolerance;
+
+        }
+
+        private static void SetAllergyIntoleranceCategory(this AllergyIntolerance allergyIntolerance, CodableValue allergenType, Extension allergyExtension)
+        {
+            foreach (CodedValue aValue in allergenType)
+            {
+                List<AllergyIntoleranceCategory?> lstAllergyIntoleranceCategory = new List<AllergyIntoleranceCategory?>();
+                AllergyIntoleranceCategory allergyIntoleranceCategory;
+                if (aValue.Value != null)
+                {
+                    if (Enum.TryParse(aValue.Value, true, out allergyIntoleranceCategory))
+                    {
+                        lstAllergyIntoleranceCategory.Add(allergyIntoleranceCategory);
+                    }
+                    else
+                    {
+                        allergyExtension.AddExtension(HealthVaultExtensions.AllergenType, new FhirString(aValue.Value));
```

Comment:
We're possibly adding multiple `allergen-type` extensions into the `allergy` extension. Is this expected?